pandas/tests/arrays/string_/test_string.py

```diff
@@ -269,3 +267,31 @@ def test_value_counts_na():
     result = arr.value_counts(dropna=True)
     expected = pd.Series([2, 1], index=["a", "b"], dtype="Int64")
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize("func", ["min", "max"])
+@pytest.mark.parametrize("skipna", [True, False])
+def test_reduction(func):
+    strings = ["x", "y", "z"]
+    arr = pd.Series(strings, dtype="string")
+    ser = pd.Series(strings)
+
+    result = getattr(arr, func)(skipna=skipna)
+    expected = getattr(ser, func)(skipna=skipna)
+
+    assert result == expected
```

Comment:
Instead of comparing to object dtype behavior, I would maybe rather hardcode the expected result, or use numpy for the expected behaviour

Reply:
Still some test failures I need to look at (I think they're tests that expect methods not to be implemented). Is the general approach here valid or a bit too hacky? I suppose a better technique would just be to implement these methods for StringArray directly, but maybe do that in a future PR?

Reply:
> Is the general approach here valid or a bit too hacky? I suppose a better technique would just be to implement these methods for StringArray directly, but maybe do that in a future PR?

I think both options are fine for me (but eventually, yes, we should also fix the methods on the array class as well)

Reply:
I think the reason why the NA-handling wasn't working was due to an apparently long-standing bug in `nanops.nanminmax` which I think we can fix here: https://github.com/pandas-dev/pandas/issues/18588. Basically we are filling NA with infinite values when taking the min or max, but this doesn't make sense for object dtypes and an error gets raised even if `skipna` is `True`.

If we fix that by explicitly masking the missing values instead, I believe we can just use this function directly in `StringArray` methods.